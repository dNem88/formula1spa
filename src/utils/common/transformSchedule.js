function transformSchedule(props) {
    let transformed = {}
    let months = ['january', 'february', 'march', 'april', 'may', 'june', 'july', 'august', 'september', 'october', 'november', 'december']
    
    if (props !== null & props !== undefined) {
        let raceDate = props.date.split('-');
        let raceStart = [raceDate[0], Number(raceDate[1])-1, Number(raceDate[2]), ...props.time.split(':').slice(0,2)]
        let UTCinMs = Date.UTC(...raceStart) /*in UTC Time*/
        transformed.raceStartUTC = UTCinMs
        let date1 = new Date(props.date);
        date1.setDate(date1.getDate() - 2)
        let weekendStart = `${date1.getDate()} ${months[date1.getMonth()].substring(0,3).toUpperCase()} ${date1.getFullYear()}`
        transformed.country = props.Circuit.Location.country.toUpperCase()
        let date = props.date.split('-');
        transformed.date = props.date;
        transformed.month = months[Number(date[1])-1].substring(0,3).toUpperCase();
        transformed.day = Number(date[2])
        transformed.imageBase = props.Circuit.Location.country.toLowerCase().split(' ').join('_')
        transformed.raceName = props.raceName;
        transformed.season = Number(props.season);
        transformed.weekend = `${weekendStart} - ${transformed.day} ${transformed.month} ${props.season}`;
        return transformed
    } else {
        return props
    }
}   

export default transformSchedule