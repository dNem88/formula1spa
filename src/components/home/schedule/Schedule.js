import React, {Fragment} from 'react'
import styles from './schedule.module.css'
import Flag from '../../common/flag/Flag'
import transformSchedule from '../../../utils/common/transformSchedule'
import ActiveSchedule from '../activeSchedule/ActiveSchedule'

function Schedule(props) {
    const transformedProps = transformSchedule(props);
    let classname = 'main-container'
    if (props._id === props.activeId) {
        classname = 'active-main-container'
    } else {
        classname = 'main-container'
    }
 
    return (
        <div id={props._id} className={styles[classname]} onClick={props.clickHandler}>
            {classname === 'main-container' ? 
                <Fragment>
                    <Flag classname={'image-container'} imageBase={transformedProps.imageBase}/>
                    <h3 className={styles.country}>{transformedProps.country}</h3>
                    <h1 className={styles.day}>{transformedProps.day}</h1>
                    <h3 className={styles.month}>{transformedProps.month}</h3>
                </Fragment> : <ActiveSchedule {...transformedProps} classname={'left-flag'}/>}
            
        </div>
    )
}

export default Schedule