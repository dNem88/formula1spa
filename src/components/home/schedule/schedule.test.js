import React from 'react'
import renderer from 'react-test-renderer'
import Schedule from './Schedule'
import {MemoryRouter} from 'react-router-dom'
import transformSchedule from '../../../utils/common/transformSchedule'


const mockSameId = {
    _id: 'test-id',
    activeId: 'test-id'
}
const mockDifferentId = {
    _id: 'test-id',
    activeId: 'diff-id'
}
let mockProps = {
    date: '05-05-2022',
    Circuit: {
        Location: {
            country: 'Bulgaria'
        }
    },
    raceName: "Imola",
    season: "2022"
}
jest.fn(transformSchedule(mockProps))
jest.mock('../activeSchedule/ActiveSchedule', () => () => {
    let Mock = 'ActiveScheduleMock'
    return <Mock/>
})

describe('Test Schedule', () => {
    it('Schedule active', () => {
        let schedule = renderer.create(
            <MemoryRouter>
                <Schedule {...mockProps} {...mockSameId}/>
            </MemoryRouter>
        ).toJSON()
        expect(schedule).toMatchSnapshot()
        expect(schedule.props.className).toBe('active-main-container')
        expect(schedule.props.id).toBe('test-id')
        
    })
    it('Schedule non-active', () => {
        let scheduleNon = renderer.create(
            <MemoryRouter>
                <Schedule {...mockDifferentId} {...mockProps}/>
            </MemoryRouter>
        ).toJSON()
        expect(scheduleNon).toMatchSnapshot()
        expect(scheduleNon.props.className).toBe('main-container')
        expect(scheduleNon.props.id).toBe('test-id')
        expect(scheduleNon.children.length).toBe(4)
    })
})