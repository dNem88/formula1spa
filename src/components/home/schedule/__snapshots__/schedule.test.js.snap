// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Schedule Schedule active 1`] = `
<div
  className="active-main-container"
  id="test-id"
>
  <ActiveScheduleMock />
</div>
`;

exports[`Test Schedule Schedule non-active 1`] = `
<div
  className="main-container"
  id="test-id"
>
  <div
    className="image-container"
  >
    <img
      alt="bulgaria-flag"
      loading="lazy"
    />
  </div>
  <h3
    className="country"
  >
    BULGARIA
  </h3>
  <h1
    className="day"
  >
    2022
  </h1>
  <h3
    className="month"
  >
    MAY
  </h3>
</div>
`;
